### PR TITLE
Adds a LICENSE file and deletes individual references to license

### PR DESCRIPTION
**Pull request by jcarres-mdsol**
Changes as in the title.
The original files did have Copyright Twitter. Given the files have changed so much probably nothing is left from those times, I do not know what to do with that @adriancole  please advise.

@jfeltesse-mdsol  @ssteeg-mdsol  @ykitamura-mdsol  FYI.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+   APPENDIX: How to apply the Apache License to your work.
+
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
```
Comment (jfeltesse-mdsol) [inline LICENSE]: I know it says otherwise in that appendix but can't we just keep the section below and call it a day??
I see literally nowhere people copy/pasting the entire license but I see the section below all the time. I'm not a lawyer but It clearly states and links to the license itself, isn't that good enough?

**Comment by jfeltesse-mdsol** [comment]
If you have access to the settings of the repo you can additionally set the license type so that it appears on the top page of the repo.

File: LICENSE (hunk 1 above)
Comment (adriancole) [inline LICENSE]: we don't have to pretend we know which file includes which copyrights. A common punt is to simply say: `Copyright 2015-2016 The OpenZipkin Authors`

**Comment by jcarres-mdsol** [comment]
I have no access to settings, probably @adriancole has

**Comment by adriancole** [comment]
@jcarres-mdsol try now :P

**Comment by jcarres-mdsol** [comment]
it seems it is more about github detecting the file: https://help.github.com/articles/adding-a-license-to-a-repository/

File: LICENSE (hunk 1 above)
Comment (jcarres-mdsol) [inline LICENSE]: I just copied this file from the openzipkin/zipkin repo.
I am good about adding a `NOTICE` file. I really dislike the header in the files because what I look for there is some comment to help me understand the code, it's frustrating to find a bunch of text which is no helpful at all.

**Comment by adriancole** [comment]
currently, we don't have any requirement to put license headers in any
files (we aren't a part of some foundation with such a mandate). I've
certainly done projects that have no license headers, except a license file
in the root dir. It is also possible to add them as a part of the
distribution (ex adding them to source only in the gem).

TL;DR; no problem from my side if you'd like to scrub the headers, just add
a NOTICE with the names of the files which originally had one. Maybe also
add a CONTRIBUTING file or similar that implies that pull requests imply
ASL licensing.

File: LICENSE (hunk 1 above)
Comment (jcarres-mdsol) [inline LICENSE]: I've added a super small notice file

File: NOTICE.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+This repository was originally developed by Twitter and released under Apache license.
```
Comment (adriancole) [inline NOTICE.md]: LG
